application/client.core/src/app/environment/components/sidebar/shell/input/component.ts

```diff
@@ -7,35 +7,13 @@ import { FormControl } from '@angular/forms';
 import { Subscription, Observable } from 'rxjs';
 import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
 import { ShellService } from '../services/service';
-import { SidebarAppShellEnvironmentComponent } from '../environment/component';
 import { Session } from '../../../../controller/session/session';
 
 import EventsSessionService from '../../../../services/standalone/service.events.session';
-import ContextMenuService, { IMenuItem } from '../../../../services/standalone/service.contextmenu';
-import PopupsService from '../../../../services/standalone/service.popups';
 import TabsSessionsService from '../../../../services/service.sessions.tabs';
 
 import * as Toolkit from 'chipmunk.client.toolkit';
 
-interface IEditing {
-    variable: boolean;
-    value: boolean;
-}
-
-export interface IEnvironment {
-    variable: string;
-    value: string;
-    custom: boolean;
-    editing: IEditing;
-    selected: boolean;
-}
-
-export interface INewInformation {
-    shell: string;
-    pwd: string;
-    env: IEnvironment[];
-}
-
 @Component({
     selector: 'app-sidebar-app-shell-input',
     templateUrl: './template.html',
```

Comment:
As soon as you are trying to get access (in styles) to outside node (`mat-autocomplete` would be created outside) you should switch your component to `encapsulation: ViewEncapsulation.None`. Also you need additional class name to recognize your options
```
<mat-option *ngFor="let recent of _ng_recent | async"
                        [value]="recent"
                        [attr.title]="recent"
                        class="recent-requests small t-console t-small app-sidebar-app-shell-input"  <---- here "app-sidebar-app-shell-input"
                        (mouseenter)="_ng_onMouseOver(recent.description)"
                        (mouseleave)="_ng_onMouseOver('')">
                <div class="options">
                    <span class="filename" [innerHTML]="_ng_getSafeHTML(recent.tdescription)"></span>
                    <mat-icon (click)="_ng_onRemove(recent.description)">close</mat-icon>
                </div>
            </mat-option>
```
And in style.less

```
mat-option.app-sidebar-app-shell-input {
    & .options {
        display: flex;
        flex-direction: row;
        align-items: center;
...
```

Reply:
please use some *class name* for parent of `& .options`. As soon as you declare it as global we should take care about possible conflicts by classnames